libopenage/unit/attribute.h

```diff
@@ -114,6 +115,7 @@ class AttributeContainer {
 
 using attr_map_t = std::map<attr_type, std::shared_ptr<AttributeContainer>>;
 
+using typeamount_set = std::map<int, unsigned int>;
```

Comment:
What is that? Please write a docstring. I guess it is the attack class to damage map.

Use `std::unordered_map` please, the `std::map` doesn't hash. And it is not a `set`, is it?